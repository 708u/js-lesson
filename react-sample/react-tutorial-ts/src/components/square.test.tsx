import React from 'react';
import Square, { SquareBtn } from './square';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('square test', () => {
  it('should render as expected components', () => {
    const component = renderer.create(
      <Square value={'X'} onClick={() => {}}/>
    )
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('has button', () => {
    const buttonWrapper = shallow(<Square value={'X'} onClick={() => {}}/>).find(SquareBtn);
    expect(buttonWrapper).toHaveLength(1);
  })
})

