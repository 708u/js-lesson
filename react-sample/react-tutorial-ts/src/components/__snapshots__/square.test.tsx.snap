// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`square test should render as expected components 1`] = `
<button
  className="sc-bdfBwQ iZlURE"
  id="btn-undefined"
  onClick={[Function]}
>
  X
</button>
`;
